tools/CoqMakefile.in

```diff
@@ -724,9 +726,14 @@ $(addsuffix .d,$(MLPACKFILES)): %.mlpack.d: %.mlpack
 	$(SHOW)'COQDEP $<'
 	$(HIDE)$(COQDEP) $(OCAMLLIBS) -c "$<" $(redir_if_ok)
 
+# If this makefile is created using a _CoqProject we have coqdep get
+# options from it. This avoids argument length limits for pathological
+# projects. Note that extra options might be on the command line.
+VDFILE_FLAGS:=$(if @PROJECT_FILE@,-f @PROJECT_FILE@,) $(CMDLINE_COQLIBS) $(CMDLINE_VFILES)
+
 $(VDFILE).d: $(VFILES)
 	$(SHOW)'COQDEP VFILES'
-	$(HIDE)$(COQDEP) $(COQLIBS) -dyndep var -c $(VFILES) $(redir_if_ok)
+	$(HIDE)$(COQDEP) -dyndep var $(VDFILE_FLAGS) $(redir_if_ok)
```

Comment:
Note here I removed the `-c` "-c : Also print the dependencies of caml modules (=ocamldep)." option. As far as I can tell coq_makefile2 started using it for .v files (probably because copy-paste).
Because coqdep looks at all files from coqproject it would have been redundant with the rules from the previous lines.

Alternatively we could try to have generated Makefiles do all dependencies in 1 coqdep command but I'm not sure if coqdep's internal ocamldep works properly and in any case that's beyond the scope of this PR.